packages/aws-rfdk/lib/deadline/scripts/python/test/test_configure_identity_registration_settings.py

```diff
@@ -0,0 +1,375 @@
+            self.assertEqual(subject.cidr_to_ipv4_match(cidr), ipv4)
+
+
+class TestConfigureIdentityRegistrationSettingsCRUD(unittest.TestCase):
+    def setUp(self):
+        self.dl_secrets = Mock()
+
+        # All these tests are set up to already have these identity registration settings applied
+        self.prior_rfdk_reg_settings_json = [ RFDK_REG_SETTING_JSON_1, RFDK_REG_SETTING_JSON_2 ]
+        self.prior_rfdk_reg_settings = [
+            RFDK_REG_SETTING_1, # Uses subnet_id="subnet-aaaaaaa"
+            RFDK_REG_SETTING_2 # Uses subnet_id="subnet-bbbbbbb"
+        ]
+
+    def test_get(self):
+        '''
+        Makes sure that a call to `get_rfdk_registration_settings` returns all the currently applied identity
+        registration settings that were created by RFDK and excludes the non-RFDK setting
+        '''
+        # GIVEN
+        configured_reg_settings = self.prior_rfdk_reg_settings_json.copy()
+        configured_reg_settings.append(NON_RFDK_REG_SETTING_JSON)
+        self.dl_secrets.run_json = MagicMock(return_value=configured_reg_settings)
+
+        # WHEN
+        rfdk_managed_settings = subject.get_rfdk_registration_settings(self.dl_secrets)
+
+        # THEN
+        self.assertListEqual(
+            rfdk_managed_settings,
+            self.prior_rfdk_reg_settings
+        )
+
+    def test_create(self):
+        '''
+        Calls `create_and_update_settings` a connection subnet ID that's used in one of the current registration
+        settings and a source subnet ID that is used in a different registration setting. Since the two subnets aren't
+        used together in a registration setting, a new one should be created.
+        '''
+        # GIVEN
+        connection_subnet_ids = [
+            "subnet-aaaaaaa" # Used by REG_SETTING_RFDK_1
+        ]
+        source_subnets = [
+            subject.SourceSubnet(subnet_id="subnet-2222222", role="Client", registration_status="Registered"), # Used by REG_SETTING_RFDK_2,
+        ]
+        subnet_to_cidr = {
+            "subnet-aaaaaaa": "111.111.0.0/16", # "111.111.*.*" Used by REG_SETTING_RFDK_1
+            "subnet-2222222": "3.3.0.0/18", # "3.3.0-63.*" # Used by REG_SETTING_RFDK_2
+        }
+
+        # WHEN
+        subject.create_and_update_settings(self.dl_secrets, self.prior_rfdk_reg_settings, connection_subnet_ids, source_subnets, subnet_to_cidr)
+
+        # THEN
+        expected = subject.LoadBalancerIdentityRegistrationSetting.from_json({
+            "ConnectionIpFilterType":"IPv4Match",
+            "ConnectionIpFilterValue":"111.111.*.*", # Similar to REG_SETTING_1
+            "SourceIpFilterType":"IPv4Match",
+            "SourceIpFilterValue":"3.3.0-63.*", # Similar to REG_SETTING_2
+            "SettingsId":"",
+            "SettingsName":"RfdkSubnet|subnet-aaaaaaa|subnet-2222222",
+            "IsEnabled":True,
+            "DefaultStatus":"Registered",
+            "DefaultRole":"Client"
+        })
```

Comment:
How about:

```suggestion
        expected = subject.LoadBalancerIdentityRegistrationSetting(
            connection_ip_filter_type   = "IPv4Match",
            connection_ip_filter_value  = subject.cidr_to_ipv4_match(subnet_to_cidr[connection_subnet_ids[0]]),
            source_ip_filter_type       = "IPv4Match",
            source_ip_filter_value      = subject.cidr_to_ipv4_match(subnet_to_cidr[source_subnets[0].subnet_id]),
            settings_id                 = "", # Ignored
            settings_name               = f"RfdkSubnet|{connection_subnet_ids[0]}|{source_subnets[0].subnet_id}",
            is_enabled                  = True,
            default_status              = source_subnets[0].registration_status,
            default_role                = source_subnets[0].role
        )
```

we already have separate testing for cidr_to_ipv4_match so this should be okay and helps illustrate the expected behavior.

What do you think?

Reply:
Updated